### PR TITLE
Remove global state from RzCons dietline

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Remove global state from dietline.

**Test plan**

Try to use `yank` actions in various modes in the Rizin command line.

See commands in https://book.rizin.re/basic_commands/dietline.html
In particular, commands like `Ctrl-k`, `Ctrl-w`, `Ctrl-y` in Emacs mode and `dw`, `d$`, `p` in the Vi mode.

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/276

### CODE REVIEW

**Review by wargio — APPROVED** [review]

File: librz/cons/dietline.c [hunk 1]
```diff
@@ -88,7 +86,7 @@ static void kill_word(BreakMode mode) {
 	I.buffer.length -= len;
 }
 
-static void paste(void) {
+static void paste(bool enable_yank_pop) {
```
Comment (ret2libc) [inline librz/cons/dietline.c]: in this function `enable_yank_pop` is set to true, but by moving it as an argument, the change to the variable is lost at the end of the function.

**Review by ret2libc — CHANGES_REQUESTED** [review]

File: librz/cons/dietline.c (hunk 1 above)
Comment (XVilka) [reply librz/cons/dietline.c]: Good catch, now I pass pointers everywhere instead!

**Review by ret2libc — APPROVED** [review]

**Review by ret2libc — COMMENTED** [review]
Only one thing... maybe the property shall go in RzCons?

**Comment by XVilka** [comment]
@ret2libc I thought about that, it's unclear where it should exactly go - RzCons structure already has too many parameters, there are two contexts - global and input ones. Maybe we should introduce a new context?

**Comment by ret2libc** [comment]
> @ret2libc I thought about that, it's unclear where it should exactly go - RzCons structure already has too many parameters, there are two contexts - global and input ones. Maybe we should introduce a new context?

I have no idea eheh i'm not that familiar with the rzcons module to know that. I'm ok with this approach for now until someone has the time/will to really review the whole module.

**Review by ret2libc — APPROVED** [review]
OK. CTRL-w/CTRL-y seem a bit broken anyway, but it's unrelated to your changes.

**Pull request merged**